src/merge.py

```diff
@@ -11,12 +11,18 @@ def merge(x: list[int], y: list[int]) -> list[int]:
     >>> merge([1, 2, 4, 6], [1, 3, 4, 5])
     [1, 1, 2, 3, 4, 4, 5, 6]
     """
-    i, j = 0, 0
-    z = []  # a new list to copy elements into
-    # FIXME: fill out the loop so you merge the lists
-    # until one of them is empty
+    i, j = 0,0 
+    z = []
     while i < len(x) and j < len(y):
-        break  # FIXME: you shouldn't just break here
-    # At least one of the lists is empty now. Copy the
-    # remainder of the other into z.
-    return z
+        if x[i] < y[j]: 
+            z.append(x[i])
+            i += 1
+        else:
+            z.append(y[j])
+            j += 1
+    if i >= len(x):
+        z.append(y[j:])
+        return z
+    else:
+        z.extend(x[i:])
+        return z
```

Comment:
The first case, you `append` when you should be using `extend`.

You don't actually have to test for the lengths here. You can `extend` an empty list just fine. So you could just do

```python
z.extend(y[j:])
z.extend(x[i:])
return z
```